### PR TITLE
Add pip check to workflows

### LINKED ISSUE #39: Add pip check to CI workflows
The current CI workflows and local test and lint commands do not catch dependency incompatibilities.

For example, installing the repository for development on the `master` branch results in these errors:

```
ERROR: autopep8 1.6.0 has requirement pycodestyle>=2.8.0, but you'll have pycodestyle 2.7.0 which is incompatible.
```
A pip check command should be made part of the local and CI tests to make sure that our dependency tree is always clean.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #39

### CODE REVIEW

**Review by katxiao — COMMENTED** [review]
Should we add a unit test workflow? [These](https://github.com/sdv-dev/DeepEcho/blob/master/tests/test_sequences.py) seem like unit tests.

**Comment by pvk-developer** [comment]
> Should we add a unit test workflow? [These](https://github.com/sdv-dev/DeepEcho/blob/master/tests/test_sequences.py) seem like unit tests.

I think they are more `integration` than `unit`, what do you think @csala ? I agree with moving them to `integration` or `unit` .

**Comment by amontanez24** [comment]
> > Should we add a unit test workflow? [These](https://github.com/sdv-dev/DeepEcho/blob/master/tests/test_sequences.py) seem like unit tests.
> 
> I think they are more `integration` than `unit`, what do you think @csala ? I agree with moving them to `integration` or `unit` .

I think I see these as unit tests as well, but either way we should add them in before merging this

File: Makefile [hunk 1]
```diff
@@ -109,9 +103,13 @@ fix-lint: ## fix lint issues using autoflake, autopep8, and isort
 
 # TEST TARGETS
 
-.PHONY: test-pytest
-test-pytest: ## run all the tests using pytest
-	invoke pytest
+.PHONY: test-integration
```
Comment (csala) [inline Makefile:106]: This should be called `test-unit` right? Below it is also invoking `integration` where it should be `unit`

**Comment by csala** [comment]
> > > Should we add a unit test workflow? [These](https://github.com/sdv-dev/DeepEcho/blob/master/tests/test_sequences.py) seem like unit tests.
> > 
> > 
> > I think they are more `integration` than `unit`, what do you think @csala ? I agree with moving them to `integration` or `unit` .
> 
> I think I see these as unit tests as well, but either way we should add them in before merging this

I agree with @amontanez24 and @katxiao  these are `unit` tests. You moved them to the right place @pvk-developer

**Review by katxiao — APPROVED** [review]
lgtm!

File: tasks.py [hunk 2]
```diff
@@ -9,8 +9,22 @@
...
+def integration(c):
+    c.run('python -m pytest ./tests/integration --reruns 3')
+
+
+@task
+def unit(c):
+    c.run('python -m pytest ./tests/unit --reruns 3')
+
+
+def coverage(c):
```
Comment (amontanez24) [inline tasks.py:26]: why doesn't this one have the @task?

**Review by amontanez24 — APPROVED** [review]
One small question but lgtm!

**Pull request merged**